common/predictive-text/build.sh

```diff
@@ -1,53 +1,62 @@
-#! /bin/bash
+#!/bin/bash
 #
 # Compiles the Language Modeling Layer for common use in predictive text and autocorrective applications.
 # Designed for optimal compatibility with the Keyman Suite.
 #
 
-display_usage ( ) {
-  echo "build.sh [-clean]"
-  echo
-  echo "  -clean              to erase pre-existing build products before a re-build"
+EMBEDDED_WORKER=embedded_worker.js
+
+# Build the worker and the main script.
+build ( ) {
+  # Build worker first; the main file depends on it.
+  # Then wrap the worker; Then build the main file.
+  npm run build-worker && wrap_worker && npm run build-main
 }
 
-# Fails the build if a specified file does not exist.
-assert ( ) {
-  if ! [ -f $1 ]; then
-    fail "Build failed."
-    exit 1
+# Creates embedded_worker.js. Must be run after the worker is built for the
+# first time
+wrap_worker ( ) {
+  echo "> wrap_worker_code LMLayerWorkerCode ./worker/index.js > ${EMBEDDED_WORKER}"
+  wrap_worker_code LMLayerWorkerCode ./worker/index.js > "${EMBEDDED_WORKER}"
+}
+
+# A nice, extensible method for -clean operations.  Add to this as necessary.
+clean ( ) {
+  rm -f "${EMBEDDED_WORKER}"
+  if [ $? -ne 0 ]; then
+    fail "Failed to erase the prior build."
   fi
 }
 
+display_usage ( ) {
+  echo "Usage: $0 [-clean] [-test | -tdd]"
+  echo "       $0 -help"
+  echo
+  echo "  -clean              to erase pre-existing build products before a re-build"
+  echo "  -help               displays this screen and exits"
+  echo "  -tdd                skips dependency updates, builds, then runs unit tests only"
+  echo "  -test               runs unit and integration tests after building"
+}
+
 # Prints a nice, common error message.
 fail ( ) {
+  local ERROR_RED
+  local NORMAL
+  ERROR_RED="$(tput setaf 1)"
```

Comment:
A thought occurs... I may have been a bit lazy when writing the original build script version.  In the base level folder "resources", we have a `shellHelperFunctions.sh` that can be included in scripts like this one.  `fail()` is already defined there, along with a few more potentially-helpful error/warning reporting functions.  It'd need a bit of change to match what you have here, but maybe take a look and see if that'd be useful?

Not a required point to advance this PR; just a suggestion.